nipoppy/workflows/dataset_status.py

```diff
@@ -0,0 +1,225 @@
+"""Workflow for init command."""
+
+import logging
+from pathlib import Path
+from typing import Optional
+
+import numpy as np
+import pandas as pd
+from rich import box
+from rich.console import Console
+from rich.table import Table
+
+from nipoppy.env import (
+    BIDS_SESSION_PREFIX,
+    BIDS_SUBJECT_PREFIX,
+    LogColor,
+    StrOrPathLike,
+)
+from nipoppy.tabular.bagel import STATUS_SUCCESS, Bagel
+from nipoppy.tabular.doughnut import Doughnut
+from nipoppy.tabular.manifest import Manifest
+from nipoppy.workflows.base import BaseWorkflow
+
+
+class StatusWorkflow(BaseWorkflow):
+    """Workflow for status command."""
+
+    def __init__(
+        self,
+        dpath_root: Path,
+        fpath_layout: Optional[StrOrPathLike] = None,
+        logger: Optional[logging.Logger] = None,
+        dry_run: bool = False,
+        save_status_to_disk: bool = False,
+    ):
+        """Initialize the workflow."""
+        super().__init__(
+            dpath_root=dpath_root,
+            name="init",
+            fpath_layout=fpath_layout,
+            logger=logger,
+            dry_run=dry_run,
+        )
+        self.fname_readme = "README.md"
+        self.save_status_to_disk = save_status_to_disk  # TODO maybe
+        self.status_df = pd.DataFrame()
```

Comment:
Edge case but I am wondering if it would be better to define the empty `status_df` here or in a `run_setup` method. Not sure...

Reply:
fixed!